File: Python/Data Structures & Algorithms Module - Python/Week4/LinkedStackApp.py
"""
LinkedStackApp.py
Billy Cussen
12/11/2020
"""

from LinkedStack import LinkedStack

myLinkedStackApp = LinkedStack()

print("Empty: "+str(myLinkedStackApp.isEmpty()))
print("Size: "+str(myLinkedStackApp.mySize()))
try:
    print("Peek: "+str(myLinkedStackApp.peek()))
except Exception as e:
    print(e)
try:
    print("Pop: "+str(myLinkedStackApp.pop()))
except Exception as e:
    print(e)
try:
    print(myLinkedStackApp.toString())
except Exception as e:
    print(e)

myLinkedStackApp.push(2)
myLinkedStackApp.push(4)
myLinkedStackApp.push(6)
myLinkedStackApp.push(8)
myLinkedStackApp.push(10)

print("Empty: "+str(myLinkedStackApp.isEmpty()))
print("Size: "+str(myLinkedStackApp.mySize()))
try:
    print("Peek: "+str(myLinkedStackApp.peek()))
except Exception as e:
    print(e)
try:
    print("Pop: "+str(myLinkedStackApp.pop()))
except Exception as e:
    print(e)
try:
    print(myLinkedStackApp.toString())
except Exception as e:
    print(e)

while myLinkedStackApp.isEmpty() == False:
    myLinkedStackApp.pop()

print("Empty: "+str(myLinkedStackApp.isEmpty()))
print("Size: "+str(myLinkedStackApp.mySize()))
try:
    print("Peek: "+str(myLinkedStackApp.peek()))
except Exception as e:
    print(e)
try:
    print("Pop: "+str(myLinkedStackApp.pop()))
except Exception as e:
    print(e)
try:
    print(myLinkedStackApp.toString())
except Exception as e:
    print(e)

File: Java/Data Structures & Algorithms Module - Java/Week8/Revision/Week4/LinkedStackApp.java
/*
LinkedStackApp.java
Billy Cussen
12/11/2020
*/
public class LinkedStackApp{

    public static void main(String [] args){
        StackInterface<Integer> myLinkedStack = new LinkedStack<Integer>();

        System.out.println("\n***LINKED STACK***\n");
        System.out.println("Is Empty: "+myLinkedStack.isEmpty());
        try{
            System.out.println("Size: "+myLinkedStack.size());
        }catch(Exception e){
            System.out.println(e.getMessage());
        }
        try{
            System.out.println("Peek: "+myLinkedStack.peek());
            }catch(Exception e){
                System.out.println(e.getMessage());
        }
        try{
            System.out.println("Pop: "+myLinkedStack.pop());
            }catch(Exception e){
                System.out.println(e.getMessage());
        }
        System.out.println(myLinkedStack.toString()+"\n");

        myLinkedStack.push(2);
        myLinkedStack.push(4);
        myLinkedStack.push(6);
        myLinkedStack.push(8);
        myLinkedStack.push(10);
        System.out.println("Is Empty: "+myLinkedStack.isEmpty());
        try{
            System.out.println("Size: "+myLinkedStack.size());
        }catch(Exception e){
            System.out.println(e.getMessage());
        }
        try{
            System.out.println("Peek: "+myLinkedStack.peek());
            }catch(Exception e){
                System.out.println(e.getMessage());
        }
        try{
            System.out.println("Pop: "+myLinkedStack.pop());
            }catch(Exception e){
                System.out.println(e.getMessage());
        }
        System.out.println(myLinkedStack.toString());

        while(!myLinkedStack.isEmpty()){
            myLinkedStack.pop();
        }
        System.out.println("Is Empty: "+myLinkedStack.isEmpty());
        try{
            System.out.println("Size: "+myLinkedStack.size());
        }catch(Exception e){
            System.out.println(e.getMessage());
        }
        try{
            System.out.println("Peek: "+myLinkedStack.peek());
            }catch(Exception e){
                System.out.println(e.getMessage());
        }
        try{
            System.out.println("Pop: "+myLinkedStack.pop());
            }catch(Exception e){
                System.out.println(e.getMessage());
        }
        System.out.println(myLinkedStack.toString());
    }
}